docs/architecture/default-eventing-for-application.md

```diff
@@ -0,0 +1,27 @@
+# Default eventing for application
+
+## Introduction
+
+The Director as the main component which manages the information for Applications and Runtimes is responsible also for providing eventing configurations. The following document describes the process of determining which runtime should be configured to process the Applications' events.
+
+> **NOTE** Currently only one Runtime can receive events from the Application
+
+Application can obtain the eventing configuration from the Directors API. It is done via the `application(id: ID!): Application`. The `Application` type offers the `eventingConfiguration` property which holds the `defaultURL` property that contains the URL to the eventing backend.
+If the value of `defaultURL` is an empty string, it means that there is no Runtime assigned that can process the Application events.
+
+## Determining default Runtime for the Application events
+
+When querying `eventingConfiguration` for the `Application` type the Director component tries to determine the default Runtime that will process the events.
+In the first step, it looks whether there is a Runtime that has the following label assigned `{APPLICATION_ID}/defaultEventing = true` and the runtime belongs to one of the application scenarios. If there is no such runtime, the Director looks for the oldest Runtime that is assigned to the Application scenarios. After having found the oldest Runtime, it is labeled with the label `{APPLICATION_ID}/defaultEventing = true` and next time it will be treated as the default one. Having the default Runtime determined, the Director component fetches its eventing URL and returns as `defaultURL` for the `eventingConfiguration`.
```

Comment:
```suggestion
In the first step, it searches for a Runtime that belongs to one of the Application's scenarios and has the `{APPLICATION_ID}/defaultEventing = true` label assigned. If there is no such Runtime, the Director looks for the oldest Runtime that is assigned to the Application's scenarios. After finding the oldest Runtime, it is labeled with the `{APPLICATION_ID}/defaultEventing = true` label and next time, it will be treated as the default one. Having the default Runtime determined, the Director component fetches its eventing URL and returns as **defaultURL** for the **eventingConfiguration**.
```